### PR TITLE
TaskRunner: repeat task if there is a recording error

### PR DESCRIPTION
**Pull request by tam-nguyen-senseye**
This resolves a potential issue that can occur when the TaskRunner encounters a recording error during a task, which will cause the runner to stay stuck on the task screen. This fix makes it so that the runner resets to the camera preview screen, essentially restarting the task.

Note: this is separate from [ch18657], which is concerned with restarting a task when a video recording is successful but an interruption was detected (e.g. user navigated away from app)

### CODE REVIEW

**Review by bobbys-dev — APPROVED** [review]

File: src/components/TaskRunner.tsx [hunk 1]
```diff
@@ -54,22 +60,47 @@ const TaskRunner: React.FunctionComponent<TaskRunnerProps> = (props) => {
...
+    (index, name, instructions) => {
+      if (onTaskPreview) {
+        onTaskPreview(index, name, instructions);
+      }
+    },
+    [onTaskPreview]
+  );
+
+  const _onTaskReset = React.useCallback(() => {
+    if (onTaskReset) {
```
[the remark this answers is not in the document]
Comment (tam-nguyen-senseye) [reply src/components/TaskRunner.tsx:79]: yes, that's correct. i intend it to also be used when implementing ch18657, thus why it's not called onRecordingError() or something more specific.

In the same vain as `onTaskPreview`, this is mainly so users can add in a custom dialog notifying the participant that they'll be redoing a task

File: src/components/TaskRunner.tsx [hunk 2]
```diff
@@ -120,7 +136,7 @@ const TaskRunner: React.FunctionComponent<TaskRunnerProps> = (props) => {
 
   // execute onEnd callback once all tasks are complete
   React.useEffect(() => {
-    if (taskIndex >= children.length && !isRecording) {
+    if (!isRecording && taskIndex >= children.length) {
```
Comment (Haplo-Dragon) [inline src/components/TaskRunner.tsx:139]: Is this a short circuit evaluation thing, or just 'cause it reads nicer? Either way :+1:

**Review by Haplo-Dragon — APPROVED** [review]

File: src/components/TaskRunner.tsx (hunk 2 above)
Comment (tam-nguyen-senseye) [reply src/components/TaskRunner.tsx:139]: the latter 😆 

the original order was because i was afraid of race conditions, but turns out it doesn't matter

**Comment by tam-nguyen-senseye** [comment]
bundling these changes with #37

**Pull request closed without merging**